### PR TITLE
feat: create sidebar design

### PR DESCRIPTION
**Pull request by yuribodo**
- add sidebar design based on the neetcode website

### CODE REVIEW

File: app/components/Sidebar.tsx [hunk 1]
```diff
@@ -1,7 +1,48 @@
+'use client';
+
+import { useState } from "react";
+
 export function Sidebar() {
+  const [activeTab, setActiveTab] = useState("algoritmos");
```
Comment (SouzaGabriel26) [inline app/components/Sidebar.tsx]: What do you think in work with types here?

This `useState` could be typed like:

```suggestion

type ActiveTabs = 'algorithms' | 'courses';

export function Sidebar() {
  const [activeTab, setActiveTab] = useState<ActiveTabs>("algorithms");
```

File: app/components/Sidebar.tsx [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
   return (
-    <aside className="w-1/6 bg-gray-800/95 rounded mt-1 text-white text-center py-2">
-      Select Roadmap
+    <aside className="w-1/5 bg-gray-800/95 rounded mt-1 text-white text-center py-2">
+      <h2 className="mt-2 font-bold">Select Roadmap</h2>
+      <div className="flex justify-center items-center space-x-6 mt-4">
+        <div
+          className={`cursor-pointer px-4 py-2 rounded-3xl font-bold text-md ${
+            activeTab === "algoritmos" ? "bg-[#4814b0] text-white" : ""
+          }`}
```
Comment (SouzaGabriel26) [inline app/components/Sidebar.tsx]: This kind of styling with tailwindcss can be made easier using some useful libraries like: [`twMerge`](https://www.npmjs.com/package/tailwind-merge) and [`clsx`](https://www.npmjs.com/package/clsx).

But for now it's good. When I set up the eslint in the project, I will add this `util function` to help us :+1:

**Review by SouzaGabriel26 — DISMISSED** [review]

File: app/components/Sidebar.tsx (hunk 2 above)
Comment (yuribodo) [reply app/components/Sidebar.tsx]: Ok. I will take a look at those libs

File: app/components/Sidebar.tsx (hunk 1 above)
Comment (yuribodo) [reply app/components/Sidebar.tsx]: Hadn't thought of it. I think that's a great idea

**Review by SouzaGabriel26 — APPROVED** [review]

**Pull request merged**